packages/protocol/test/stability/exchange.ts

```diff
@@ -626,6 +705,57 @@ contract('Exchange', (accounts: string[]) => {
             })
           })
 
+          describe('when stableBucket needs to be capped', () => {
+            beforeEach(async () => {
+              await registry.setAddressFor(CeloContractName.Exchange, owner)
+
+              await mockSortedOracles.setMedianRate(
+                stableToken.address,
+                new BigNumber('0x20000000000000000')
+              ) // 1 CELO = 2 USD
+
+              await stableToken.mint(user, new BigNumber('61e24')) // 6M
+              await registry.setAddressFor(CeloContractName.Exchange, exchange.address)
+            })
+
+            it('has the right stableBucketFractionCap', async () => {
+              assertEqualBN(await exchange.stableBucketFractionCap(), toFixed(1 / 22))
+            })
+
+            it("doesn't hit the cap and it shouldn't be resized", async () => {
+              const [tradeableGold, mintableStable] = await exchange.getBuyAndSellBuckets(false)
+              assertEqualBN(mintableStable, new BigNumber('2e21'))
+              assertEqualBN(tradeableGold, new BigNumber('1e21'))
+
+              // the buckets hold the price
+              assertEqualBN(
+                new BigNumber(mintableStable.dividedBy(tradeableGold)),
+                new BigNumber('2')
+              )
+            })
+
+            it('has the correct getStableBucketCap', async () => {
+              assertEqualBN(
+                await exchange.getStableBucketCap(),
+                new BigNumber('2772727272727272816000000')
+              )
+            })
+
+            it('has the right price after the cap', async () => {
```

Comment:
I would probably make this a another test group and set the rate in the `before` of the group and make separate assertions in separate tests. Right now it looks like you are just checking that the price/ratio of the buckets?